app/plugin.go

```diff
@@ -270,6 +270,13 @@ func (a *App) ShutDownPlugins() {
 
 	a.RemoveConfigListener(a.Srv.PluginConfigListenerId)
 	a.Srv.PluginConfigListenerId = ""
+
+	// Acquiring lock manually before cleaning up PluginsEnvironment.
+	a.Srv.PluginsLock.Lock()
+	defer a.Srv.PluginsLock.Unlock()
+	if a.Srv.PluginsEnvironment != pluginsEnvironment {
+		mlog.Warn("Another PluginsEnvironment detected while shutting down plugins.")
```

Comment:
What does this warning message actually say? If I as a admin see this, what does it tell me? Same questions as a developer.

Reply:
It means that the system ran into an unexpected situation while shutting down plugins and further investigation is needed to figure out why that error occured. Similar to the demo plugin forcefully being shutdown because it took longer than 10 seconds to shutdown.

Reply:
> It means that the system ran into an unexpected situation while shutting down plugins and further investigation is needed to figure out why that error occured. 

How would this investigation look like? What can/should do an admin do?

> Similar to the demo plugin forcefully being shutdown because it took longer than 10 seconds to shutdown.

That message is targeting plugin devs, which makes sense to me.

Reply:
> How would this investigation look like? What can/should do an admin do

Don't think admin would be able to do much unfortunately. It's more for MM devs (mostly us I suppose) to investigate who's starting a new PluginsEnv when shutting down the current one.

Reply:
What do you think about lowering it to `debug` in this case? If anyone besides dev should ignore this, it should not be warning IMO.

Reply:
Can having multiple PluginsEnv considered a harmful event? if the answer to that question is yes then I think this should be left a warning. I don't see it being harmful to the system 🤷‍♂, so I'm ok changing this to `debug`, but not sure what others think.